classes/04class/exercises/c04-iac02/jayanath/vpc.tf

```diff
@@ -0,0 +1,164 @@
+# Create the VPC first
+resource "aws_vpc" "doa_iac_vpc" {
+  cidr_block = var.vpc_cidr
+
+  tags = {
+    Name = "devopsacademy-iac"
+  }
+}
+
+# A NACL for our new VPC
+# The from_port and to_port values will be ignored as the protocol value is set to all
+resource "aws_network_acl" "main" {
+  vpc_id = aws_vpc.doa_iac_vpc.id
+
+  egress {
+    protocol   = "all"
+    rule_no    = 100
+    action     = "allow"
+    cidr_block = "10.0.0.0/16"
+    from_port  = 0
+    to_port    = 0
+  }
+
+  egress {
+    protocol   = "all"
+    rule_no    = 200
+    action     = "allow"
+    cidr_block = "121.200.5.90/32"
+    from_port  = 0
+    to_port    = 0
+  }
+
+  ingress {
+    protocol   = "all"
+    rule_no    = 300
+    action     = "allow"
+    cidr_block = "10.0.0.0/16"
+    from_port  = 0
+    to_port    = 0
+  }
+
+  ingress {
+    protocol   = "all"
+    rule_no    = 400
+    action     = "allow"
+    cidr_block = "121.200.5.90/32"
+    from_port  = 0
+    to_port    = 0
+  }
+
+  tags = {
+    Name = "iac-vpc-nacl"
+  }
+}
+
+# Create IGW and attach to the VPC
+resource "aws_internet_gateway" "iac_vpc_gw" {
+  vpc_id = aws_vpc.doa_iac_vpc.id
+
+  tags = {
+    Name = "iac-vpc-igw"
+  }
+}
+
+# Create subnets
+resource "aws_subnet" "private_a" {
+  vpc_id            = aws_vpc.doa_iac_vpc.id
+  cidr_block        = var.subnets.private_subnet_a.cidr
+  availability_zone = var.subnets.private_subnet_a.az
+  tags = {
+    Name = var.subnets.private_subnet_a.name
+  }
+}
+
+resource "aws_subnet" "private_b" {
+  vpc_id            = aws_vpc.doa_iac_vpc.id
+  cidr_block        = var.subnets.private_subnet_b.cidr
+  availability_zone = var.subnets.private_subnet_b.az
+  tags = {
+    Name = var.subnets.private_subnet_b.name
+  }
+}
+
```

Comment:
as you have `var.subnets` as a map, you coulkd use `for_each` here to easily iterate through it and simplify your code.